fplc_method_validator.py

```
@@ -0,0 +1,497 @@
+                elif step_type == StepType.GRADIENT:
+                    parsed_steps.append(MethodGradientStep(
+                        step_type=step_type,
+                        duration=step_data['duration'],
+                        flow_rate=step_data['flow_rate'],
+                        start_gradient=step_data['start'],
+                        end_gradient=step_data['end'],
+                        collection=step_data['collection']
+                    ))
+            except (KeyError, ValueError) as e:
+                logging.error(f"Error parsing step {i+1} in template '{name}': {e}. Skipping this step or template.")
+                raise TemplateError(f"Failed to parse template '{name}' due to invalid step data at step {i+1}: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)